src/main/java/by/dragonsurvivalteam/dragonsurvival/client/handlers/magic/ClientMagicHandler.java

```diff
@@ -105,7 +105,7 @@ public static void renderEffectParticle(LivingEntity entity, ParticleOptions dat
 
 	@SubscribeEvent
 	@OnlyIn( Dist.CLIENT )
-	public static void removeLavaAndWaterFog(ViewportEvent.RenderFog event){
+	public static void removeLavaAndWaterFog(ViewportEvent.RenderFog event){ // TODO :: Check for fluid first
```

Comment:
Just as a note: this part of the code is most likely created for lava vision compatibility with shaders.

Reply:
the comment is in reference to `event.getCamera().getFluidInCamera()`

checking for that first instead of getting the capability at the start would probably be a performance improvement (not a noticable one id assume though)

Reply:
The best is the enemy of the good, but there is no limit to perfection.